importer.py

```diff
@@ -1113,6 +1118,66 @@ class EggNode:
     pass
 
 
+class EggDistanceNode(EggNode):
+
+    default_vertex = EggVertex((0, 0, 0))
+
+    def __init__(self):
+        self.vertex = EggDistanceNode.default_vertex
+
+    def begin_child(self, context, type, name, values):
+        type = type.upper()
+        if type != 'VERTEX':
+            assert False
+
+        return EggVertex(tuple(parse_number(v) for v in values))
+
+    def end_child(self, context, type, name, child):
+        self.vertex = child
+
+    def build_tree(self, context, parent=None, inv_matrix=None, under_dart=False):
+        parent.vertex = self.vertex
+
+
+class EggSwitchNode(EggNode):
+
+    default_vertex = EggVertex((0, 0, 0))
+
+    def __init__(self):
+        self.vertex = EggSwitchNode.default_vertex
+        self.fade = False
+        self.distance = ()
+
+    def begin_child(self, context, type, name, values):
+        type = type.upper()
+
+        if type == 'DISTANCE':
+            self.distance = tuple(parse_number(v) for v in values[:2])
+            
+            if len(values) == 3: # eggs do not support fade yet but it is in manual
+                self.fade = bool(values[2])
```

Comment:
Out of curiosity, what do you mean by "eggs do not support fade yet"?

Why is this a boolean? The value is float in Panda.

Reply:
According the [manual entry for egg switch conditions](https://docs.panda3d.org/1.10/python/tools/model-export/egg-syntax?highlight=egg#switchcondition) 
> “fade” is presently ignored.

Trying to convert a egg with a fade specified doesn't seem to work in my testing either. I will however change this to a float that is my error.